src/Base/UnitsSchema.cpp

```diff
@@ -21,29 +21,31 @@
  ***************************************************************************/
 
 
-#include "PreCompiled.h"
-#ifdef __GNUC__
-#include <unistd.h>
-#endif
-
-#include <QLocale>
-#include <QString>
+#include <iomanip>
+#include <sstream>
 
 #include "UnitsSchema.h"
 
-
 using namespace Base;
 
-QString
-UnitsSchema::toLocale(const Base::Quantity& quant, double factor, const QString& unitString) const
+std::string UnitsSchema::toLocale(const Base::Quantity& quant,
+                                  double factor,
+                                  const std::string& unitString) const
 {
-    QLocale Lc;
+    std::stringstream ss;

```

Comment:
With this implementation the NumberOptions of QuantityFormat are not respected any more. And the created string is locale-independent.

Reply:
I see. Here's where `ss.imbue(std::locale());` should be.

Reply:
Somehow this doesn't work as it should. This part is responsible for the failure of the content of the quantity spin boxes and the incorrect dimension string of the 3D viewer.

For testing purposes I have restored the old implementation but converting the string from std::string to QString and back and this fixes the problems.
```
std::string UnitsSchema::toLocale(const Base::Quantity& quant,
                                  double factor,
                                  const std::string& unitString) const
{
    QLocale Lc;
    const QuantityFormat& format = quant.getFormat();
    if (format.option != QuantityFormat::None) {
        int opt = format.option;
        Lc.setNumberOptions(static_cast<QLocale::NumberOptions>(opt));
    }

    QString Ln = Lc.toString((quant.getValue() / factor), format.toFormat(), format.precision);
    return QString::fromUtf8("%1 %2").arg(Ln, QString::fromStdString(unitString)).toStdString();
}
```

If it's OK for you we can use this as a tmp. solution until we have figured out how to do it correctly with I/O streams.

Reply:
Sure it is okay. What locale are using for testing? de-DE?

Reply:
Tested with `ss.imbue(std::locale("de_DE.UTF8"));` and it seems correct, so I have to figure out how is FreeCAD working with locales.
Temporary solution is of course okay, if we are in hurry merging this. Otherwise I'd rather look inside this deeper.

Reply:
FreeCAD seems to expect locale stuff happens at Qt level only. At least `Translator::setLocale` cares only about `QLocale` (should add something like `std::locale::global(...)`)

Reply:
As I have no idea how to easily solve above problem, let's stick with original implementation for now. Comment was added into code.

Reply:
I should have read C++ reference more carefully. `std::numpunct_byname` does the trick. I'm temped to put this PR on hold and fix `std::locale` issue first. @wwmayer, it that okay for you?

Reply:
> If it's OK for you we can use this as a tmp. solution until we have figured out how to do it correctly with I/O streams.

@wwmayer, the templates in `CommandT.h` are using `boost::format` which [renders all parameters using global locale](https://www.boost.io/doc/libs/1_67_0/libs/locale/doc/html/localized_text_formatting.html), so we cannot switch locale globally. I will revert locale changes and go back to your proposal.